app/controllers/studies_controller.rb

```diff
@@ -741,7 +743,11 @@ def update_study_file_from_sync
           @study.default_options[:cluster] = @study_file.name
           @study.save
         end
+        old_name = @cluster.name.dup
         @cluster.update(name: @study_file.name)
+        # also update data_arrays, using declared indices rather than Mongoid association for better performance
+        DataArray.where(study_id: @study.id, study_file_id: @cluster.study_file.id, cluster_name: old_name,
+                        linear_data_type: 'ClusterGroup', linear_data_id: @cluster.id).update_all(cluster_name: @study_file.name)
```

Comment:
maybe pull the query part of this into a class method on DataArray.  Something like
 ```
def self.find_for_cluster(cluster)
   return DataArray.where(study_id: @study.id, study_file_id: @cluster.study_file.id, cluster_name: old_name,
                        linear_data_type: 'ClusterGroup', linear_data_id: @cluster.id)
end
```

Reply:
73c287e